# Inhibitory vs Exitatory

### creline reporter positive only

In [1]:
%matplotlib inline

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

### Data load & check

In [3]:
data = pd.read_csv('cell_data.csv')
data.head()

line_name  specimen__id                         specimen__name  \
0   Scnn1a-Tg2-Cre     313862022    Scnn1a-Tg2-Cre;Ai14-167081.03.02.01   
1   Rorb-IRES2-Cre     479508018  Rorb-IRES2-Cre-D;Ai14-196662.03.01.01   
2  Htr3a-Cre_NO152     442627661   Htr3a-Cre_NO152;Ai14-178904.06.01.01   
3     Cux2-CreERT2     487358945      Cux2-CreERT2;Ai14-214155.05.01.01   
4   Scnn1a-Tg3-Cre     469803127    Scnn1a-Tg3-Cre;Ai14-180215.05.02.01   

  specimen__hemisphere  structure__id               structure__name  \
0                right            721  Primary visual area, layer 4   
1                 left            721  Primary visual area, layer 4   
2                 left      312782566       Anterior area, layer 6a   
3                 left            778  Primary visual area, layer 5   
4                right            778  Primary visual area, layer 5   

  structure__acronym  structure_parent__id structure_parent__acronym  \
0              VISp4                   385                      VISp   
1              VISp4                   385                      VISp   
2             VISa6a             312782546                      VISa   
3              VISp5                   385                      VISp   
4              VISp5                   385                      VISp   

  structure__layer ...    cell_reporter_status  m__glif  \
0                4 ...   cre reporter positive        0   
1                4 ...   cre reporter positive        2   
2               6a ...   cre reporter positive        0   
3                5 ...   cre reporter positive        5   
4                5 ...   cre reporter positive        5   

   m__biophys_perisomatic  m__biophys_all_active  tag__apical  \
0                       1                      0       intact   
1                       0                      0       intact   
2                       0                      0          NaN   
3                       0                      0       intact   
4                       0                      0       intact   

   tag__dendrite_type                            morph_thumb_path  \
0               spiny  /api/v2/well_known_file_download/487659609   
1               spiny                                         NaN   
2              aspiny                                         NaN   
3               spiny                                         NaN   
4               spiny                                         NaN   

                             ephys_thumb_path  \
0  /api/v2/well_known_file_download/487623088   
1  /api/v2/well_known_file_download/491384619   
2  /api/v2/well_known_file_download/487610447   
3  /api/v2/well_known_file_download/487361512   
4  /api/v2/well_known_file_download/487635954   

                 ephys_inst_thresh_thumb_path   id  
0  /api/v2/well_known_file_download/491376069  606  
1  /api/v2/well_known_file_download/491384621  522  
2  /api/v2/well_known_file_download/491377923   81  
3  /api/v2/well_known_file_download/491767690   38  
4  /api/v2/well_known_file_download/491376335  671  

[5 rows x 44 columns]

In [5]:
len(data)

692

### Whole data fearture 44, observation 692

In [6]:
data_cre = data[data['cell_reporter_status'] == 'cre reporter positive']
len(data_cre)

615

### cre positive cell 615

### cre reporter posivitive cells tagging.

In [19]:
def dis_EI(cre_line): # labelling for excitory vs inbihitory cre line
    output = []
    for item in cre_line:
        if (item == 'Pvalb-IRES-Cre' or item == 'Sst-IRES-Cre' or
            item == 'Htr3a-Cre_NO152' or item == 'Vip-IRES-Cre' or
            item == 'Gad2-IRES-Cre'):
                output = output + ['I']
        else:
            output = output + ['E']
    return output
ExIn = []
ExIn = dis_EI(data_cre['line_name']) # this is python list!!
ExIn_pd = pd.DataFrame({'ExIn': ExIn}) # making pandas frame
data_cre = data_cre.join(ExIn_pd)


ValueError: columns overlap but no suffix specified: Index(['ExIn'], dtype='object')

## subsetting data - only electrophysiological component for comparision

In [23]:

data_sub = data_cre.iloc[:,17:27]
data_sub = data_sub.join(data_cre.iloc[:, 0])
data_sub = data_sub.join(data_cre.iloc[:, 44])
data_sub.head()

ef__upstroke_downstroke_ratio_long_square  ef__adaptation  \
0                                   5.261166        0.288858   
1                                   4.957590        0.097034   
2                                   4.843871        0.026744   
3                                   4.815331             NaN   
4                                   4.756384        0.022992   

   ef__f_i_curve_slope  ef__threshold_i_long_square    ef__tau  ef__avg_isi  \
0             0.159756                        190.0  17.653078   147.790000   
1             0.042555                        130.0  21.290832   106.357500   
2             0.122172                        230.0   6.668957   112.607500   
3             0.006781                        130.0  22.450481   140.730000   
4             0.245112                         50.0  28.245981    57.503437   

   ef__avg_firing_rate      ef__ri  ef__peak_t_ramp  ef__vrest  \
0             6.766358   58.281287         2.552270 -63.574978   
1             9.402252  189.375005         2.475857 -72.237417   
2             8.880403  106.250000        12.004272 -78.260158   
3             7.105805  151.562640         2.939443 -77.838111   
4            17.390265  199.375002         2.150840 -76.375813   

         line_name ExIn  
0   Scnn1a-Tg2-Cre    E  
1   Rorb-IRES2-Cre    E  
2  Htr3a-Cre_NO152    I  
3     Cux2-CreERT2    E  
4   Scnn1a-Tg3-Cre    E

### value counts

In [24]:
line_name_freq = pd.DataFrame(data_sub['line_name'].value_counts())
line_name_freq

line_name
Pvalb-IRES-Cre          139
Sst-IRES-Cre            108
Htr3a-Cre_NO152          78
Rorb-IRES2-Cre           70
Rbp4-Cre_KL100           49
Cux2-CreERT2             44
Nr5a1-Cre                40
Scnn1a-Tg3-Cre           33
Ntsr1-Cre_GN220          19
Scnn1a-Tg2-Cre           15
Gad2-IRES-Cre            11
Slc17a6-IRES-Cre          3
Ntsr1-Cre                 2
Ctgf-2A-dgCre             2
Vip-IRES-Cre              2

In [26]:
ExIn_freq = pd.DataFrame(data_sub['ExIn'].value_counts())
ExIn_freq

ExIn
I   308
E   242

In [34]:
data_sub_E = data_sub[data_sub['ExIn'] == 'E']
len(data_sub_E)

242

In [35]:
data_sub_I = data_sub[data_sub['ExIn'] == 'I']
len(data_sub_I)

308

In [36]:
data_sub_E.mean(0)

ef__upstroke_downstroke_ratio_long_square      3.463708
ef__adaptation                                 0.061771
ef__f_i_curve_slope                            0.175076
ef__threshold_i_long_square                  129.173553
ef__tau                                       19.402332
ef__avg_isi                                   88.390272
ef__avg_firing_rate                           13.937327
ef__ri                                       164.553576
ef__peak_t_ramp                                4.318287
ef__vrest                                    -74.339143
dtype: float64

In [37]:
data_sub_I.mean(0)

ef__upstroke_downstroke_ratio_long_square      1.974728
ef__adaptation                                 0.042603
ef__f_i_curve_slope                            0.436578
ef__threshold_i_long_square                  158.490259
ef__tau                                       16.754750
ef__avg_isi                                   45.242125
ef__avg_firing_rate                           43.495115
ef__ri                                       188.491038
ef__peak_t_ramp                                6.336382
ef__vrest                                    -71.953926
dtype: float64

In [40]:
data_sub_E.std(0)

ef__upstroke_downstroke_ratio_long_square     0.547509
ef__adaptation                                0.068820
ef__f_i_curve_slope                           0.084239
ef__threshold_i_long_square                  74.255214
ef__tau                                       8.296157
ef__avg_isi                                  48.023374
ef__avg_firing_rate                           7.186785
ef__ri                                       59.686916
ef__peak_t_ramp                               2.363933
ef__vrest                                     4.785310
dtype: float64

In [41]:
data_sub_I.std(0)

ef__upstroke_downstroke_ratio_long_square      0.713841
ef__adaptation                                 0.076055
ef__f_i_curve_slope                            0.370664
ef__threshold_i_long_square                  123.178890
ef__tau                                       10.838854
ef__avg_isi                                   40.207535
ef__avg_firing_rate                           38.633299
ef__ri                                        95.048258
ef__peak_t_ramp                                4.747852
ef__vrest                                      5.263367
dtype: float64

In [43]:
data_sub.describe()

/home/hyunsu/anaconda3/lib/python3.5/site-packages/numpy/lib/function_base.py:3823: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


ef__upstroke_downstroke_ratio_long_square  ef__adaptation  \
count                                 615.000000      502.000000   
mean                                    2.470953        0.046840   
std                                     1.037507        0.072595   
min                                     0.862726       -0.204913   
25%                                     1.447692             NaN   
50%                                     2.459218             NaN   
75%                                     3.346059             NaN   
max                                     5.261166        0.586461   

       ef__f_i_curve_slope  ef__threshold_i_long_square     ef__tau  \
count           615.000000                   615.000000  615.000000   
mean              0.391070                   164.658536   16.776006   
std               0.392468                   122.713473    9.966241   
min              -0.015048                    10.000000    3.467032   
25%               0.145817                    70.000000    8.356607   
50%               0.220765                   130.000000   15.918908   
75%               0.498809                   220.000000   22.312366   
max               2.382667                   800.000000   90.353511   

       ef__avg_isi  ef__avg_firing_rate      ef__ri  ef__peak_t_ramp  \
count   573.000000           573.000000  615.000000       579.000000   
mean     60.449486            36.389130  170.094651         6.110615   
std      50.028697            39.240043   82.205505         4.628768   
min       3.828333             1.976792   36.562462         0.284310   
25%            NaN                  NaN  111.796895              NaN   
50%            NaN                  NaN  154.374963              NaN   
75%            NaN                  NaN  206.796877              NaN   
max     505.870000           261.210274  570.312446        28.140622   

        ef__vrest  
count  615.000000  
mean   -73.136684  
std      5.111332  
min    -85.449162  
25%    -76.837734  
50%    -73.601509  
75%    -70.243455  
max    -50.776325